homeassistant/components/evohome/__init__.py

```diff
@@ -685,10 +685,6 @@ def _dt_evo_to_aware(dt_naive: dt, utc_offset: timedelta) -> dt:
 
     async def _update_schedule(self) -> None:
         """Get the latest schedule, if any."""
-        if "DailySchedules" in self._schedule and not self._schedule["DailySchedules"]:
-            if not self._evo_device.setpointStatus["setpointMode"] == EVO_FOLLOW:
-                return  # avoid unnecessary I/O - there's nothing to update
-
         self._schedule = await self._evo_broker.call_client_api(
             self._evo_device.schedule(), refresh=False
         )
```

Comment:
The exception handler returns None when this API call fails (both before and after being the handler is refactored in #43678) - testing has revealed that the code raise an unhandled exception in this case.